storage/Devices/BlkDeviceImpl.cc

```diff
@@ -825,4 +867,46 @@ namespace storage
 	}
     }
 
+
+    void
+    wait_for_detach_devices(const vector<const BlkDevice*>& blk_devices)
+    {
+	vector<string> dev_names;
+
+	for_each(blk_devices.begin(), blk_devices.end(), [&dev_names](const BlkDevice* dev)
```

Comment:
Why not use a simple `for` loop here? You are iterating over the entire container from begin() to end().
I might be overlooking something here, but this seems awfully complicated for just collecting those names.